src/net/README.md

```diff
@@ -83,76 +75,71 @@ and generate the `next` generation.  However, as mentioned previously,
+So, the user (sysrepo) needs to add `cmd.exit` scripts in the current
+tree for every leaf that leaves a parent, and `cmd.init` scripts for
+leaves that are new or added to parents in the next generation.
 
 In our example, interface `eth4` is moved from `lag0` to `br0`, so we
-need to run `eth4/ip-link.dn` in the current generation first to remove
-`eth4` from `lag0` before its `ip-link.up` script in the next generation
+need to run `eth4/ip.exit` in the current generation first to remove
+`eth4` from `lag0` before its `ip.init` script in the next generation
 sets `eth4` as a bridge member instead.
 
-We traverse the current generation and execute all `.dn` scripts:
+We traverse the current generation and execute all `cmd.exit` scripts:
 
     /run/net/<GEN>/
-	   |-- lo/
-	   |    `-- deps/
 	   |-- br0/
 	   |    |-- deps/
 	   |    |    |-- eth1 -> ../../eth1
 	   |    |    |-- eth2 -> ../../eth2
 	   |    |    |-- eth3 -> ../../eth3
 	   |    |    `-- lag0 -> ../../lag0
-	   |    `-- ip-link.up
+	   |    `-- ip.init
 	   |-- eth0/
 	   |    |-- deps/
-	   |    |-- ip-link.up
-	   |    `-- ip-addr.up
+	   |    `-- ip.init
 	   |-- eth4/
 	   |    |-- deps/
-	   |    |-- ip-link.dn
-	   |    `-- ip-link.up
+	   |    |-- ip.exit
+	   |    `-- ip.init
 	   |-- lag0/
 	   |    |-- deps/
 	   |    |    |-- eth4 -> ../../eth4
 	   |    |    `-- eth5 -> ../../eth5
-	   |    `-- ip-link.up
+	   |    `-- ip.init
 	   `-- vlan1/
 	        |-- deps/
 	        |    `-- br0 -> ../../br0
-	        |-- ip-link.up
-	        `-- ip-addr.up
+	        `-- ip.init
 
-Now we can run all the `.up` scripts in the next generation:
+Now we can run all the `cmd.init` scripts in the next generation:
 
     /run/net/<GEN+1>/
-	   |-- lo/
-	   |    |-- deps/
-	   |    |-- ip-link.up
-	   |    `-- ip-addr.up
 	   |-- br0/
 	   |    |-- deps/
 	   |    |    |-- eth1 -> ../../eth1
 	   |    |    |-- eth2 -> ../../eth2
 	   |    |    |-- eth3 -> ../../eth3
 	   |    |    |-- eth4 -> ../../eth4
 	   |    |    `-- lag0 -> ../../lag0
-	   |    `-- ip-link.up
+	   |    `-- ip.init
 	   |-- eth0/
 	   |    |-- deps/
-	   |    |-- ip-link.up
-	   |    `-- ip-addr.up
+	   |    `-- ip.init
 	   |-- eth4/
 	   |    |-- deps/
-	   |    |-- ip-link.dn
-	   |    `-- ip-link.up
+	   |    `-- ip.init
 	   |-- lag0/
 	   |    |-- deps/
 	   |    |    `-- eth5 -> ../../eth5
-	   |    `-- ip-link.up
+	   |    `-- ip.init
 	   `-- vlan1/
 	        |-- deps/
 	        |    `-- br0 -> ../../br0
-	        |-- ip-link.up
-	        `-- ip-addr.up
-
-
+	        `-- ip.init
+
+When there are no changes compared to the previous generation, the
+`cmd.init` scripts can be empty.  The existence of an `ip.init` script,
+however, means that it is allowed to be brought up on `net up`.  The
+existence of a directory (named after an interface) means the interface
+should be brought down on `net down`.  Doing `net up ifname` is a subset
+of that for a single interface `ifname`.
```

Comment:
What about maintaining a `/run/net/$GEN/$IFACE/admin-state`?

Possible values:
- `disabled`: Can only be set by sysrepo. `net` is not ever allowed to bring it up
- `up`: Set by sysrepo for enabled interfaces, and by `net up` after a previous `net down`
- `down`: Set by `net down` to temporarily bring down an interface that was previously in the `up` state

Reply:
Great suggestion, much cleaner.  I'll push fix!